### PR TITLE
EZP-30962: Dropped deprecated permission-related methods

### PR DESCRIPTION
**Pull request by ViniTou**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-30962](https://jira.ez.no/browse/EZP-30962)
| **Bug/Improvement**| no
| **New feature**    | no
| **Target version** | `master`
| **BC breaks**      | yes
| **Tests pass**     | yes
| **Doc needed**     | yes

This is first batch of removed methods, mostly deprecated stuff from Repository layer. For easier review, you can go by commits.

Green build:
https://travis-ci.org/ezsystems/ezpublish-kernel/builds/585548905

**TODO**:
- [x] Implement feature / fix a bug.
- [x] Implement tests.
- [x] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php [hunk 1]
```diff
@@ -61,7 +61,8 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
         /** @var \eZ\Publish\API\Repository\Repository $repository */
         $repository = $this->getContainer()->get('ezpublish.api.repository');
-        $repository->setCurrentUser(
+        $permissionResolver = $this->getContainer()->get('eZ\Publish\API\Repository\PermissionResolver');
```
Comment (adamwojs) [inline eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]:
```suggestion
        $permissionResolver = $this->getContainer()->get(PermissionResolver::class);
```

or just inject permission resolver into command.

Comment (ViniTou) [reply eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]: Not sure if this a correct way, as `eZ\Publish\API\Repository\PermissionResolver` refers to service name and not class per se.

Comment (alongosz) [reply eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]: > Not sure if this a correct way, as `eZ\Publish\API\Repository\PermissionResolver` refers to service name and not class per se.

The whole point of switching to FQCN-named services is the ability to use `::class` notation.

Comment (gggeek) [reply eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]: FWIW: I am generally not fond of FQCN-named services for platform/library code.
The whole point of using a service name instead of a class name is to abstract away the service implementation from its interface. And in any given eZ project, I do generally reimplement a few services from the Kernel, to add fixes or extra capabilities. Using the FQCN of the original class does not help me in that case, as it kinda hints that the service in use is still the original eZ one, whereas it is not any more...
I'd leave usage of QCN-named services to app code, unless there are other big advantages to that (I discount the fact that using class names helps autocomplete in IDEs as a big advantage, as autocomplete-based coding leads to frequent and sometimes subtle mistakes).

Comment (alongosz) [reply eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]: > FWIW: I am generally not fond of FQCN-named services for platform/library code.
> The whole point of using a service name instead of a class name is to abstract away the service implementation from its interface. And in any given eZ project, I do generally reimplement a few services from the Kernel, to add fixes or extra capabilities. Using the FQCN of the original class does not help me in that case, as it kinda hints that the service in use is still the original eZ one, whereas it is not any more...

Please use your own implementation FQCN name along with [`decorates`](https://symfony.com/doc/3.4/service_container/service_decoration.html) attribute inside Symfony DIC.

Comment (alongosz) [reply eZ/Bundle/EzPublishMigrationBundle/Command/LegacyStorage/UpdateSortKeysCommand.php]: Sorry for the delay, I was off.

> `decorates` is useful, but that means that my own code will be using one service name which is different from the name used by the kernel to refer to the same service. 

No, it doesn't :-)
One of the advantages of doing e.g.:
``` yaml
services:
    App\MyResolver:
        decorates: eZ\Publish\API\Repository\PermissionResolver
        arguments:
            $innerPermissionResolver: '@App\MyResolver.inner'
```
``` php
class MyResolver implements PermissionResolver
{
    /** @var \eZ\Publish\API\Repository\PermissionResolver */
    private $innerPermissionResolver;

    public function __construct(PermissionResolver $innerPermissionResolver)
    {
        $this->innerPermissionResolver = $innerPermissionResolver;
    }
    // ...
}
```
is that when injecting, either explicitly or by auto-wiring, the original service:

``` php
class MyService
{
    /** @var \eZ\Publish\API\Repository\PermissionResolver */
    private $permissionResolver;

    public function __construct(PermissionResolver $permissionResolver)
    {
        $this->permissionResolver = $permissionResolver;
    }
}
```
you'll actually inject your decorator.

File: composer.json [hunk 2]
```diff
@@ -93,7 +93,7 @@
     "extra": {
         "_ci_branch-comment_": "Keep ci branch up-to-date with master or branch if on stable. ci is never on github but convention used for ci behat testing!",
         "_ezplatform_branch_for_behat_tests_comment_": "ezplatform branch to use to run Behat tests",
-        "_ezplatform_branch_for_behat_tests": "master",
+        "_ezplatform_branch_for_behat_tests": "ezp-30934-deprecated_methods",
```
Comment (alongosz) [inline composer.json]: Friendly reminder: please drop this before sending to the QA.

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 3]
```diff
@@ -32,29 +32,34 @@ public function process(ContainerBuilder $container)
...
         $daoAuthenticationProviderDef->setClass(RepositoryAuthenticationProvider::class);
         $daoAuthenticationProviderDef->addMethodCall(
-            'setRepository',
-            [$repositoryReference]
+            'setPermissionResolver',
+            [$permissionResolverRef]
+        );
+        $daoAuthenticationProviderDef->addMethodCall(
+            'setUserService',
+            [$userServiceRef]
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]: any reason for not injecting those via DIC directly?

File: eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php [hunk 4]
```diff
@@ -32,29 +32,34 @@ public function process(ContainerBuilder $container)
         }
 
         $configResolverRef = new Reference('ezpublish.config.resolver');
-        $repositoryReference = new Reference('ezpublish.api.repository');
+        $permissionResolverRef = new Reference('eZ\Publish\API\Repository\PermissionResolver');
+        $userServiceRef = new Reference('eZ\Publish\API\Repository\UserService');
```
Comment (alongosz) [inline eZ/Bundle/EzPublishCoreBundle/DependencyInjection/Compiler/SecurityPass.php]:
``` php
use eZ\Publish\API\Repository\UserService;
// ...
$userServiceRef = new Reference(UserService::class);
```
but investigate if those setters are still needed. Maybe we just need a constructor for this class with auto-wiring enabled.

File: doc/bc/changes-8.0.md [hunk 5]
```diff
@@ -61,6 +61,16 @@ Changes affecting version compatibility with former or future versions.
     * `eZ\Publish\Core\MVC\Symfony\Matcher\ContentMatcherFactory`
     * `eZ\Publish\Core\MVC\Symfony\Matcher\LocationMatcherFactory`
 
+* Following methods have been removed:
+
+    * `\eZ\Publish\API\Repository\Values\Content\Query\Criterion::createFromQueryBuilder`
```
Comment (alongosz) [inline doc/bc/changes-8.0.md]: Seems this one is out of scope as 95% of the PR deals with authentication

**Comment by ViniTou** [comment]
@alongosz 
fixed & rebased
